File: python/day1/solution.py
import os

def main():

    elves = []

    # This is an ugly solution
    with open('./input.txt') as f:
        lines = f.read()
        
        split = lines.split(2*os.linesep)
        print(split)

        for set in split:
            values = set.split(os.linesep)
            total = 0
            for value in values:
                total += int(value)

            elves.append(total)

    print(max(elves))

if __name__ == "__main__":
    main()

File: haskell/day1/solution.hs
import Data.List

main :: IO ()
main = do
  content <- readFile ("./input.txt")
  let scores = calcCalories (lines content) 0 []
  print scores
  print $ maximum scores
  print $ sum $ map (nthLargest scores) [0, 1, 2]

-- recurse along list from left to right, restarting total when we reach a ""
calcCalories :: [String] -> Integer -> [Integer] -> [Integer]
calcCalories ("":xs) score scores = calcCalories xs 0 (score : scores)
calcCalories (x:xs) score scores = calcCalories xs (score + (read x ::Integer)) scores
calcCalories [] score scores = score : scores

nthLargest xs n = (reverse $ sort xs) !! n